jina/types/document/__init__.py

```diff
@@ -227,6 +227,8 @@ def __init__(
                 assert d.tags['good'] == 'bye'  # true
         """
         self._pb_body = jina_pb2.DocumentProto()
+        self._embedding = NdArray(self._pb_body.embedding)
```

Comment:
this will not let the user dynamically change the embedding and its type. Plus it will make every Document object take twice the memory space in terms of embedding

Reply:
I don't think this consumes twice memory because the allocated space is for the protobuf body which is the same

Reply:
ah okey, we are not getting the value yet. I see.